src/base/bittorrent/infohash.cpp

```diff
@@ -78,16 +89,56 @@ InfoHash::operator QString() const
 
 bool BitTorrent::operator==(const InfoHash &left, const InfoHash &right)
 {
-    return (static_cast<lt::sha1_hash>(left)
-            == static_cast<lt::sha1_hash>(right));
+    return (static_cast<NativeHash>(left)
+            == static_cast<NativeHash>(right));
 }
 
 bool BitTorrent::operator!=(const InfoHash &left, const InfoHash &right)
 {
     return !(left == right);
 }
 
+#if LIBTORRENT_VERSION_NUM >= 20000
+namespace std {
+    template <>
+    struct hash<lt::info_hash_t>
```

Comment:
Doesn't libtorrent provide `std::hash` for "info hash" type? Maybe there are two separate instances for sha1 and sha512?
Anyway it doesn't seem like good idea to me to define it just to be used inside qHash. I would move the logic there instead.